crypto/x509/t_req.c

```diff
@@ -129,8 +129,10 @@ int X509_REQ_print_ex(BIO *bp, X509_REQ *x, unsigned long nmflags,
                     count = X509_ATTRIBUTE_count(a);
  get_next:
                     at = X509_ATTRIBUTE_get0_type(a, ii);
-                    type = at->type;
-                    bs = at->value.asn1_string;
+                    if (at != NULL) {
+                        type = at->type;
+                        bs = at->value.asn1_string;
+                    }
                 }
                 for (j = 25 - j; j > 0; j--)
```

Comment:
potentially bs could be NULL and then dereferenced bs->length further down will now fall over.

Reply:
If in the first loop at != NULL then the type and bs will be set.
On the second loop if att is NULL then bs and type will still be set to the previous values.. Is this what you want to happen?

Reply:
I doubt it. Either NULL is invalid here and we should return (or print) some error, or it's valid and print that.

Reply:
I think that setting type to 0 (causing a diagnostic print) will be enough here.

Reply:
> The get_next: construct looks a bit funny in here (looks like a for loop :))

It looks weird, but I'm not ready to refactor this place without a relevant example.

Reply:
The problem is not that at == NULL, the problem is that count == 0.

Reply:
And that count == 0 seems to be invalid.